src/de/jost_net/JVerein/server/EinstellungImpl.java

```diff
@@ -122,6 +124,34 @@ protected void insertCheck() throws ApplicationException
         throw new ApplicationException(e.getMessage());
       }
 
+      try
+      {
+        AltersgruppenParser ap = new AltersgruppenParser(getBeitragAltersstufen());
+        ArrayList<VonBis> vbs = new ArrayList<VonBis>();
+        while(ap.hasNext())
+        {
+          vbs.add(ap.getNext());
+        }
+        for(int i=0;i<100;i++)
+        {
+          boolean found = false;
+          for(VonBis vb : vbs)
+          {
+            if(i >= vb.getVon() && i <= vb.getBis())
+            {
+              found = true;
+              break;
+            }
+          }
+          if(!found)
+            throw new ApplicationException("Keine passende Altersstufe gefunden f�r " + i + " Jahre");
+        }
+      }
+      catch (RuntimeException e)
+      {
+        throw new ApplicationException(e.getMessage());
```

Comment:
Der Altersgruppenparser liefert in Zeile 52 eine RuntimeException mit zweizeiligem Text. Unsere Anzeige hat aber nur eine Zeile und darum ist die Meldung schwer zu lesen.
Ich denke es wäre sinnvoll im Altersgruppenparser den Zeilenumbruch herauszunehmen.
Zum Test kann man einfach eine Gruppe ohne Range erzeugen also z.B. 19-99 durch 19-98,99 ersetzen.

Es ist mir aber jetzt noch etwas aufgefallen was ich mir nicht erklären kann. Setzt man in den Einstellungen die 19-98,99 am Ende und drückt Speichern, gibt es die Fehlermeldung aber die Daten werden trotzdem gespeichert. Das gleiche passiert auch in Einstellungen->Statistik so. 
Ich dachte eigentlich immer, dass nicht gespeichert wird wenn der updateCheck schief geht. Ist das ein Fehler im Code der e.store() handhabt und en updateCheck() aufruft?.

Reply:
```
Es ist mir aber jetzt noch etwas aufgefallen was ich mir nicht erklären kann. Setzt man in den Einstellungen die 19-98,99 am Ende und drückt Speichern, gibt es die Fehlermeldung aber die Daten werden trotzdem gespeichert. Das gleiche passiert auch in Einstellungen->Statistik so.
Ich dachte eigentlich immer, dass nicht gespeichert wird wenn der updateCheck schief geht. Ist das ein Fehler im Code der e.store() handhabt und den updateCheck() aufruft?.
```
Das ist mir auch schon aufgefallen ich habe aber keine passende Lösung gefunden. Beim Speichern wird erst das statische Einstellungen Object aktuallisiert und dann der UpdateCheck ausgeführt. Dadurch sind die neuen Daten nicht in der DB gespeichert aber in der aktuellen Session trotzdem so vorhanden.